packages/core/test/integration/query-interface/bulk-delete.test.ts

```diff
@@ -0,0 +1,105 @@
+import { sequelize } from '../support';
+
+const queryInterface = sequelize.queryInterface;
+
+describe('QueryInterface#bulkDelete', () => {
+  beforeEach(async () => {
+    await queryInterface.createTable('level', {
+      id: {
+        type: DataTypes.INTEGER,
+        primaryKey: true,
+        autoIncrement: true,
+      },
+      name: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+    });
+
+    await queryInterface.createTable('actors', {
+      id: {
+        type: DataTypes.INTEGER,
+        primaryKey: true,
+        autoIncrement: true,
+      },
+      name: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+      status: {
+        type: DataTypes.STRING,
+        allowNull: true,
+      },
+      age: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+      },
+      level_id: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        references: {
+          key: 'id',
+          table: 'level',
+        },
+        onDelete: 'CASCADE',
+        onUpdate: 'CASCADE',
+      },
+    });
+
+    await queryInterface.bulkInsert('level', [
+      { name: 'level1' },
+      { name: 'level2' },
+      { name: 'level3' },
+    ]);
+
+    await queryInterface.bulkInsert('actors', [
+      { name: 'actor1', status: 'active', age: 10, level_id: 1 },
+      { name: 'actor2', status: 'active', age: 20, level_id: 2 },
+      { name: 'actor3', status: 'active', age: 30, level_id: 3 },
+    ]);
+  });
+
+  it('should truncate the table', async () => {
+    await queryInterface.bulkDelete('actors', {}, { truncate: true });
+
+    const count = await queryInterface.select(null, 'actors', {});
+    expect(count).to.have.length(0);
+  });
+
+  if (sequelize.dialect.supports.truncate.cascade) {
+    it('should truncate the table with cascade', async () => {
+      await queryInterface.bulkDelete('level', {}, { truncate: true, cascade: true });
+
+      const levels = await queryInterface.select(null, 'level', {});
+      const actors = await queryInterface.select(null, 'actors', {});
+      expect(levels).to.have.length(0);
+      expect(actors).to.have.length(0);
+    });
+  }
+
+  if (sequelize.dialect.supports.truncate.restartIdentity) {
+    it('should truncate the table with restart identity', async () => {
+      await queryInterface.bulkDelete('actors', {}, { truncate: true, restartIdentity: true });
+
+      const count = await queryInterface.select(null, 'actors', {});
+      expect(count).to.have.length(0);
+
+      await queryInterface.bulkInsert('actors', [
+        { name: 'actor1', status: 'active', age: 10, level_id: 1 },
+        { name: 'actor2', status: 'active', age: 20, level_id: 2 },
+        { name: 'actor3', status: 'active', age: 30, level_id: 3 },
+      ]);
+
+      const [actor1, actor2, actor3] = await queryInterface.select(null, 'actors', {});
+
+      // @ts-expect-error -- id is not defined on the type
+      expect(actor1.id).to.equal(1);
```

Comment:
We should type these as well, I believe we have similar test suites that do this

Reply:
Done